### DISCUSSION TITLE
en/chapter_array_and_linkedlist/linked_list/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#1371] [Announcements]
# en/chapter_array_and_linkedlist/linked_list/

Data Structures and Algorithms Crash Course with Animated Illustrations and Off-the-Shelf Code

https://www.hello-algo.com/en/chapter_array_and_linkedlist/linked_list/

**Comment by bit-scientist** [comment]
Is there any reason for doing:
```
class ListNode:
    def __init__(self, val: int):
        self.val: int = val
        self.next: ListNode | None = None
```
instead of:
```
class ListNode:
    def __init__(self, val: int, next: ListNode | None = None):
        self.val = val
        self.next = next
```
Why is it the case that `next` is introduced in the constructor's body rather than as argument similar to `val`?

**Reply by lulzshadowwalker** [reply]
I would assume that to be a preference, up to you. Having a constructor parameter allows you to do the following:
```py
n = ListNode(1, ListNode(2, ListNode(3)))
```
at the end of the day it's your code, you can design your API however you see best.
